### PR TITLE
Node16 EoL

### PR DESCRIPTION
**Pull request by NSeydoux**
- Drop support for Node 16
- Upgrade dependencies

### CODE REVIEW

File: .babelrc.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (NSeydoux) [inline .babelrc.json:1]: This is auto-generated by storybooks

File: .babelrc.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          "safari": 15,
+          "firefox": 91
+        }
+      }
+    ],
+    "@babel/preset-typescript",
+    "@babel/preset-react"
+  ],
+  "plugins": []
+}
```
Comment (jeswr) [inline .babelrc.json:18]: Missing newline at end of file

File: .eslintrc.js [hunk 3]
```diff
@@ -22,6 +22,11 @@ module.exports = {
     "no-shadow": ["warn", { allow: ["fetch"] }],
     "react/require-default-props": "warn",
     "react/default-props-match-prop-types": "warn",
+    "react/function-component-definition": "off",
     noImplicitAny: "off",
+    // FIXME: re-enable the following
```
Comment (NSeydoux) [inline .eslintrc.js:27]: I'm creating a ticket for that

File: .storybook/main.js [hunk 4]
```diff
@@ -1,27 +1,27 @@
 module.exports = {
-  core: {
```
Comment (NSeydoux) [inline .storybook/main.js:2]: All these changes are managed by storybook's migration CLI

File: package.json [hunk 5]
```diff
@@ -21,71 +21,78 @@
...
-    "eslint-plugin-storybook": "^0.6.10",
-    "jest": "^29.4.1",
-    "jest-environment-jsdom": "^29.4.1",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-jest": "^27.6.0",
+    "eslint-plugin-jsx-a11y": "^6.8.0",
+    "eslint-plugin-prettier": "^5.1.1",
+    "eslint-plugin-react": "^7.33.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-storybook": "^0.6.15",
```
Comment (jeswr) [inline package.json:78]: I think most of these can be removed

**Review by jeswr — APPROVED** [review]

File: .babelrc.json (hunk 2 above)
Comment (NSeydoux) [reply .babelrc.json]: https://github.com/inrupt/solid-ui-react/pull/901/commits/01865653bae86614ceaec0443aec073e2866f320

File: package.json (hunk 5 above)
Comment (NSeydoux) [reply package.json]: https://github.com/inrupt/solid-ui-react/pull/901/commits/52fbddd0b2e6a9f94512c8736690f19dc02a6f6b

**Pull request merged**